sardine/src/srd.rs

```diff
@@ -548,6 +548,6 @@ impl Srd {
         hash.input(&self.client_nonce);
         hash.input(&self.server_nonce);
 
-        self.iv.clone_from_slice(&hash.result().to_vec()[0..16]);
+        self.iv.clone_from_slice(&hash.result().to_vec());
```

Comment:
Is it useful to keep the full 32 bits ?  I remembered it was an issue at some point...

Reply:
ChaCha uses 24 bytes and AES uses 16, so I instead kept the 32 and used what I needed in the encryption functions. I tested this and it works. We could keep 24 bytes but that would really only save 8 bytes and add an extra operation.

Reply:
Make sense 👍